packages/twenty-front/src/modules/object-record/record-board/states/recordBoardNewRecordByColumnIdComponentFamilyState.ts

```diff
@@ -1,10 +1,13 @@
+import { EntityForSelect } from '@/object-record/relation-picker/types/EntityForSelect';
 import { createComponentFamilyState } from '@/ui/utilities/state/component-state/utils/createComponentFamilyState';
 
 export type NewCard = {
   id: string;
   columnId: string;
   isCreating: boolean;
   position: 'first' | 'last';
+  isOpportunity: boolean;
+  company: EntityForSelect | null;
 };
```

Comment:
@FelixMalfait How about this? used the same state with additional keys for opportunity creation!

Reply:
Oh yes I haven't looked at the code in details again yet but that seems much better :)

Love that you removed more lines that you added 🙌
<img width="152" alt="Screenshot 2024-09-27 at 17 34 19" src="https://github.com/user-attachments/assets/b3cae418-bede-4d84-9713-177a62fc8f6f">